src/components/PupilComponents/PupilUnitsSection/PupilUnitsSection.tsx

```diff
@@ -58,6 +58,7 @@ export type PupilUnitsSectionProps = {
   applyFilter: (subjectCategory: string) => void;
   showTooltip?: boolean;
   id?: string;
+  onCallback?: (unit: UnitListingBrowseData[number]) => void;
```

Comment:
not fond of this name. Can we have something more descriptive .... onUnitSelected  ?